api/pkg/apis/v1alpha1/utils/symphony-api.go

```diff
@@ -62,29 +61,37 @@ func (e *SummarySpecError) Error() string {
 }
 
 func GetSymphonyAPIAddressBase() string {
-	if symphonyAPIAddressBase == "" {
+	if os.Getenv(constants.SymphonyAPIUrlEnvName) == "" {
 		return SymphonyAPIAddressBase
 	}
-	return symphonyAPIAddressBase
+	return os.Getenv(constants.SymphonyAPIUrlEnvName)
 }
 
-var getApiClientOnce sync.Once
-var symphonyApiClient *apiClient
-var apiClientError error
+var symphonyApiClients sync.Map
 
 func GetApiClient() (*apiClient, error) {
-	getApiClientOnce.Do(func() {
-		symphonyApiClient, apiClientError = getApiClient()
-	})
-	return symphonyApiClient, apiClientError
-}
-
-// For testing purpose only
-func UpdateApiClientUrl(url string) {
-	if symphonyApiClient == nil {
-		GetApiClient()
+	symphonyBaseUrl := os.Getenv(constants.SymphonyAPIUrlEnvName)
+	if value, ok := symphonyApiClients.Load(symphonyBaseUrl); ok {
+		client, ok := value.(*apiClient)
+		if !ok {
+			log.Infof("Symphony base url apiclient is broken. Recreating it.")
```

Comment:
L77 to L83 are almost the same as L87 to L93, suggest changing the code flow and remove the duplicated code.